dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/icon/IconController.java

```diff
@@ -124,17 +124,9 @@ public void getIconData(@PathVariable String key, HttpServletResponse response)
   }
 
   @GetMapping
-  public @ResponseBody List<IconResponse> getAllIcons(
-      @RequestParam(required = false) String[] keywords) {
-    List<Icon> icons;
-
-    if (keywords == null) {
-      icons = iconService.getIcons();
-    } else {
-      icons = iconService.getIcons(keywords);
-    }
+  public @ResponseBody List<IconResponse> getAllIcons(IconCriteria iconCriteria) {
```

Comment:
Would `getIcons` make more sense as a method name? As it does not necessarily return all icons if keywords are given. Or does all mean something else here 😅

Reply:
in cases when there are no keywords and IconType.ALL then it will return all icons.

Reply:
Right, so the method is a more general one hence `getAllIcons` is misleading as that is just one of the use cases it supports.